### PR TITLE
Add ZSTD compression

### LINKED ISSUE #35: zSTD Support
Hi, is there a reason zstd support isn't in this client? Would you be willing to accept a PR for it or is there a reason it's purposely left out?

### PR DESCRIPTION
**Pull request by byian**
To resolve #35 

- Added zstd v1.5.5 sources to contrib
- New cmake option WITH_SYSTEM_ZSTD to choose between building and using the system installed library
- Compression and decompression are implemented using the Simple API (so there is no choice of compression level)
- Updated tests

### CODE REVIEW

**Comment by byian** [comment]
@Enmk Could you please trigger CI for this?

File: clickhouse/base/compressed.cpp [hunk 1]
```diff
@@ -60,55 +59,79 @@ bool CompressedInput::Decompress() {
...
         } else {
             mem_.Reset(data_.data(), original);
         }
+        return true;
+    }
+
+    case static_cast<uint8_t>(CompressionMethodByte::NONE): {
+
+        throw CompressionError("compression method not defined" + std::to_string((method)));
+    }
```
Comment (Enmk) [inline clickhouse/base/compressed.cpp:124]: defa
```suggestion
    }
    default:
        throw CompressionError("Unknown or unsupported compression method " + std::to_string((method)));
```

**Review by Enmk — CHANGES_REQUESTED** [review]
Some modifications required

File: clickhouse/base/compressed.cpp (hunk 1 above)
Comment (byian) [reply clickhouse/base/compressed.cpp:129]: done

File: clickhouse/base/compressed.h [hunk 2]
```diff
@@ -6,6 +6,20 @@
...
 
+/// Methods of block compression.
+enum class CompressionMethod : int8_t {
+    None = -1,
+    LZ4  = 1,
+    ZSTD = 2,
+};
+
+// see DB::CompressionMethodByte from src/Compression/CompressionInfo.h of ClickHouse project
+enum class CompressionMethodByte : uint8_t {
```
[the remark this answers is not in the document]
Comment (byian) [reply clickhouse/base/compressed.h]: moved to cpp

File: clickhouse/client.h [hunk 3]
```diff
@@ -3,6 +3,8 @@
 #include "query.h"
 #include "exceptions.h"
 
+#include "base/compressed.h"
```
[the remark this answers is not in the document]
Comment (byian) [reply clickhouse/client.h]: done

**Review by Enmk — APPROVED** [review]
LGTM

**Pull request merged**